src/plugins/ui_core_views/cell_evaluation/evaluation_plugin.ts

```diff
@@ -207,10 +211,22 @@ export class EvaluationPlugin extends UIPlugin {
   // ---------------------------------------------------------------------------
 
   evaluateFormula(sheetId: UID, formulaString: string): CellValue | Matrix<CellValue> {
+    const result = this.evaluateFormulaResult(sheetId, formulaString);
+    if (isMatrix(result)) {
+      return matrixMap(result, (cell) => cell.value);
+    }
+    return result.value;
+  }
+
+  evaluateFormulaResult(
+    sheetId: UID,
+    formulaString: string
+  ): ValueAndFormat | Matrix<ValueAndFormat> {
     try {
-      return this.evaluator.evaluateFormula(sheetId, formulaString);
+      return this.evaluator.evaluateFormulaResult(sheetId, formulaString);
     } catch (error) {
-      return error instanceof EvaluationError ? error.errorType : CellErrorType.GenericError;
+      const value = error instanceof EvaluationError ? error.errorType : CellErrorType.GenericError;
+      return { value };
```

Comment:
shouldn't we return the same as the actual evaluation using `handleError` ?

Reply:
Ah si, you're right